### PR TITLE
Add a CLI flag --stats-flush-interval-duration to allow specifying stats flush interval in Duration

### PR DESCRIPTION
**Pull request by qqustc**
New flag --stats-flush-interval-duration is mutually exclusive with --stats-flush-interval.

### CODE REVIEW

**Comment by mum4k** [comment]
@eric846 please review and assign back to me once done.

File: test/options_test.cc [hunk 1]
```diff
@@ -84,11 +84,78 @@ TEST_F(OptionsImplTest, DurationAndNoDurationSanity) {
...
+      "googleapis.com/"
+      "envoy.config.metrics.v3.StatsdSink\",tcp_cluster_name:\"statsd\"}}";
+  std::unique_ptr<OptionsImpl> options =
+      TestUtility::createOptionsImpl(fmt::format("{} http://foo", client_name_));
+  EXPECT_EQ(5, options->statsFlushInterval());
+  EXPECT_EQ(0, options->statsFlushIntervalDuration().seconds());
+  EXPECT_EQ(0, options->statsFlushIntervalDuration().nanos());
+
+  CommandLineOptionsPtr cmd = options->toCommandLineOptions();
+  EXPECT_FALSE(cmd->has_stats_flush_interval_duration());
```
Comment (eric846) [inline test/options_test.cc:148]: This part could be split out into a separate test

**Review by eric846 — CHANGES_REQUESTED** [review]

File: api/client/options.proto [hunk 2]
```diff
@@ -264,10 +264,20 @@ message CommandLineOptions {
...
+    // Time interval (number of seconds) between periodical flushes to
+    // configured stats sinks. If not specified the default is 5 seconds.
+    // stats_flush_interval must be at least 1s and at most 300s.
+    google.protobuf.UInt32Value stats_flush_interval = 35
+        [(validate.rules).uint32 = {gte: 1, lte: 300}];
+
+    // Time interval in Duration between periodical flushes to configured stats
+    // sinks.
+    google.protobuf.Duration stats_flush_interval_duration = 110
+        [(validate.rules).duration.gte.nanos = 0];
```
[the remark this answers is not in the document]
Comment (qqustc) [reply api/client/options.proto]: Done

File: test/options_test.cc [hunk 3]
```diff
@@ -84,11 +84,78 @@ TEST_F(OptionsImplTest, DurationAndNoDurationSanity) {
...
+}
+
+TEST_F(OptionsImplTest, OutOfRangeStatsFlushIntervalDuration) {
+  const std::string sink_json =
+      "{name:\"envoy.stat_sinks.statsd\",typed_config:{\"@type\":\"type."
+      "googleapis.com/"
+      "envoy.config.metrics.v3.StatsdSink\",tcp_cluster_name:\"statsd\"}}";
+  EXPECT_THROW_WITH_REGEX(
+      TestUtility::createOptionsImpl(fmt::format(
+          "{} --stats-flush-interval-duration -1.000000001s --stats-sinks {} http://foo",
```
[the remark this answers is not in the document]
Comment (qqustc) [reply test/options_test.cc]: Done

File: test/options_test.cc (hunk 1 above)
Comment (qqustc) [reply test/options_test.cc:157]: Done

**Review by eric846 — APPROVED** [review]
LGTM

**Review by mum4k — APPROVED** [review]

**Comment by mum4k** [comment]
The integration tests can no longer execute due to #820. 

Patched this PR in and executed the tests locally to confirm they are passing. Going to merge as is.

**Pull request merged**